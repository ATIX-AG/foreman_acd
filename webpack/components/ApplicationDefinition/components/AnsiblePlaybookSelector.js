import React from 'react';
import PropTypes from 'prop-types';
import ExtSelect from '../../common/ExtSelect';
import RailsData from '../../common/RailsData';

const AnsiblePlaybookSelector = ({
  label,
  hidden,
  editable,
  viewText,
  selectValue,
  onChange,
  options,
  additionalData,
}) => (
  <div className="form-group">
    <label className="col-md-2 control-label">{label}</label>
    <div className="col-md-4">
      <ExtSelect
        hidden={hidden}
        editable={editable}
        viewText={viewText}
        selectValue={selectValue}
        onChange={onChange}
        options={options}
        additionalData={additionalData}
      />
      <RailsData
        key="ansible_playbook_data"
        view="app_definition"
        parameter="acd_ansible_playbook_id"
        value={selectValue}
      />
    </div>
  </div>
);

AnsiblePlaybookSelector.propTypes = {
  label: PropTypes.string.isRequired,
  editable: PropTypes.bool.isRequired,
  viewText: PropTypes.string,
  selectValue: PropTypes.string,
  hidden: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.object,
  additionalData: PropTypes.object,
};
AnsiblePlaybookSelector.defaultProps = {
  viewText: '',
  selectValue: '',
  options: undefined,
  additionalData: undefined,
};

export default AnsiblePlaybookSelector;
