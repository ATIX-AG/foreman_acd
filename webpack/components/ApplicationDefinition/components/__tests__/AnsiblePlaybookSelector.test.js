import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import AnsiblePlaybookSelector from '../AnsiblePlaybookSelector';

const noop = () => {};

const fixtures = {
  'should render ansible playbook selector': {
    label: 'myLabel',
    hidden: false,
    editable: true,
    viewText: 'myText',
    selectValue: '1',
    onChange: noop,
    options: { first: 'first', second: 'second'},
    additionalData: { moreData: 'moooore' },
  },
  'should render hidden ansible playbook selector': {
    label: 'myLabel',
    hidden: true,
    editable: true,
    viewText: 'myText',
    selectValue: '1',
    onChange: noop,
    options: { first: 'first', second: 'second'},
    additionalData: { moreData: 'moooore' },
  },
  'should render not editable ansible playbook selector': {
    label: 'myLabel',
    hidden: false,
    editable: false,
    viewText: 'myText',
    selectValue: '1',
    onChange: noop,
    options: { first: 'first', second: 'second'},
    additionalData: { moreData: 'moooore' },
  },
};

describe('AnsiblePlaybookSelector', () =>
  testComponentSnapshotsWithFixtures(AnsiblePlaybookSelector, fixtures));
