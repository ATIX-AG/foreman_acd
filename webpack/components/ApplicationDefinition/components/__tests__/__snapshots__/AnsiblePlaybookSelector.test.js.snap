// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsiblePlaybookSelector should render ansible playbook selector 1`] = `
<div
  className="form-group"
>
  <label
    className="col-md-2 control-label"
  >
    myLabel
  </label>
  <div
    className="col-md-4"
  >
    <ExtSelect
      additionalData={
        Object {
          "moreData": "moooore",
        }
      }
      editable={true}
      hidden={false}
      onChange={[Function]}
      options={
        Object {
          "first": "first",
          "second": "second",
        }
      }
      selectValue="1"
      viewText="myText"
    />
    <RailsData
      key="ansible_playbook_data"
      parameter="acd_ansible_playbook_id"
      value="1"
      view="app_definition"
    />
  </div>
</div>
`;

exports[`AnsiblePlaybookSelector should render hidden ansible playbook selector 1`] = `
<div
  className="form-group"
>
  <label
    className="col-md-2 control-label"
  >
    myLabel
  </label>
  <div
    className="col-md-4"
  >
    <ExtSelect
      additionalData={
        Object {
          "moreData": "moooore",
        }
      }
      editable={true}
      hidden={true}
      onChange={[Function]}
      options={
        Object {
          "first": "first",
          "second": "second",
        }
      }
      selectValue="1"
      viewText="myText"
    />
    <RailsData
      key="ansible_playbook_data"
      parameter="acd_ansible_playbook_id"
      value="1"
      view="app_definition"
    />
  </div>
</div>
`;

exports[`AnsiblePlaybookSelector should render not editable ansible playbook selector 1`] = `
<div
  className="form-group"
>
  <label
    className="col-md-2 control-label"
  >
    myLabel
  </label>
  <div
    className="col-md-4"
  >
    <ExtSelect
      additionalData={
        Object {
          "moreData": "moooore",
        }
      }
      editable={false}
      hidden={false}
      onChange={[Function]}
      options={
        Object {
          "first": "first",
          "second": "second",
        }
      }
      selectValue="1"
      viewText="myText"
    />
    <RailsData
      key="ansible_playbook_data"
      parameter="acd_ansible_playbook_id"
      value="1"
      view="app_definition"
    />
  </div>
</div>
`;
